captum/attr/_core/gradient_shap.py

```diff
@@ -23,15 +26,46 @@ def __init__(self, forward_func):
         """
         GradientAttribution.__init__(self, forward_func)
 
+    @typing.overload
     def attribute(
         self,
-        inputs,
-        baselines,
-        n_samples=5,
-        stdevs=0.0,
-        target=None,
-        additional_forward_args=None,
-        return_convergence_delta=False,
+        inputs: TensorOrTupleOfTensors,
+        baselines: Union[Tensor, Callable, Tuple[Tensor, ...]],
+        n_samples: int = 5,
+        stdevs: Union[float, Tuple[float, ...]] = 0.0,
+        target: Optional[
+            Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]]]
+        ] = None,
+        additional_forward_args: Any = None,
+        return_convergence_delta: bool = True,
```

Comment:
@vivekmig  little confused here. 
Should it be with default value True or False ?

Reply:
It should be with the original default value (False). In Python 3.7, we can't actually type hint based on the literal value, so the extra overload is encoding that if the return_convergence_delta argument is omitted (which leads to default False), the return value would be just the tensor. To make it clearer, this is what the first one should look like:
```
   @typing.overload
    def attribute(
        self,
        inputs: TensorOrTupleOfTensors,
        baselines: Union[TensorOrTupleOfTensors, Callable[..., TensorOrTupleOfTensors]],
        n_samples: int = 5,
        stdevs: Union[float, Tuple[float, ...]] = 0.0,
        target: Optional[
            Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]]]
        ] = None,
        additional_forward_args: Any = None,
    ) ->TensorOrTupleOfTensors:
        ...

   @typing.overload
    def attribute(
        self,
        inputs: TensorOrTupleOfTensors,
        baselines: Union[TensorOrTupleOfTensors, Callable[..., TensorOrTupleOfTensors]],
        n_samples: int = 5,
        stdevs: Union[float, Tuple[float, ...]] = 0.0,
        target: Optional[
            Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]]]
        ] = None,
        additional_forward_args: Any = None,
        return_convergence_delta: bool = False,
    ) -> Union[TensorOrTupleOfTensors, Tuple[TensorOrTupleOfTensors, Tensor]]:
        ...

    def attribute(
        self,
        inputs,
        baselines,
        n_samples = 5,
        stdevs = 0.0,
        target = None,
        additional_forward_args = None,
        return_convergence_delta = False,
    ):
```

Reply:
This may cause an issue in attr/_core/neuron/neuron_gradient_shap.py, to fix this, just change the type hint for baselines on line 45 to:
`Union[TensorOrTupleOfTensors, Callable[..., TensorOrTupleOfTensors]]`

Reply:
@vivekmig This was really helpful. I have made changes according to these suggestions.